sql: multiregion-9node-3region-3azs-vec-off query has no home region

pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test.TestCCLLogic_multi_region_remote_access_error [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9827164?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9827164?buildTab=artifacts#/) on release-23.1 @ [6636edf9d8c0ebf9cdeedeeffd563ecb49da5b8a](https://github.com/cockroachdb/cockroach/commits/6636edf9d8c0ebf9cdeedeeffd563ecb49da5b8a):


```
=== RUN   TestCCLLogic_multi_region_remote_access_error
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/875622737c3d90631dff1d5650587860/logTestCCLLogic_multi_region_remote_access_error1552426472
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
[08:09:58] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error: 17 statements
[08:10:00] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error: 35 statements
[08:10:05] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error: 40 statements
    logic.go:2871: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error:343: WITH vtab AS (SELECT * FROM messages_rbt)
        SELECT vtab.account_id, vtab.message FROM vtab
          INNER LOOKUP JOIN messages_rbr rbr on vtab.account_id = rbr.account_id and rbr.crdb_region = 'ap-southeast-2'
        expected success, but found
        (XCHR2) Query has no home region. Try adding a LIMIT clause. For more information, see https://www.cockroachlabs.com/docs/stable/cost-based-optimizer.html#control-whether-queries-are-limited-to-a-single-region
        relational.go:2118: in buildDistribute()
[08:10:05] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error with config multiregion-9node-3region-3azs-vec-off: 43 tests, 1 failures
    logic.go:3983: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error:349: error while processing
    logic.go:3983: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2245/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/multiregion-9node-3region-3azs-vec-off/multiregion-9node-3region-3azs-vec-off_test_/multiregion-9node-3region-3azs-vec-off_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error:349: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/875622737c3d90631dff1d5650587860/logTestCCLLogic_multi_region_remote_access_error1552426472
--- FAIL: TestCCLLogic_multi_region_remote_access_error (20.55s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCCLLogic_multi_region_remote_access_error.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27427

--- a/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error
+++ b/pkg/ccl/logictestccl/testdata/logic_test/multi_region_remote_access_error
@@ -340,13 +340,14 @@ SELECT
     f_random_zip_code();
 
 # A local lookup join with a CTE as input should succeed.
-query IT
+query IT retry
 WITH vtab AS (SELECT * FROM messages_rbt)
 SELECT vtab.account_id, vtab.message FROM vtab
   INNER LOOKUP JOIN messages_rbr rbr on vtab.account_id = rbr.account_id and rbr.crdb_region = 'ap-southeast-2'
 ----
 
 # A CTE built from an INSERT which must access remote rows should fail.
+retry
 statement error pq: Query has no home region\. Try adding a filter on parent\.crdb_region and/or on key column \(parent\.p_id\)\. For more information, see https://www.cockroachlabs.com/docs/stable/cost-based-optimizer.html#control-whether-queries-are-limited-to-a-single-region
 WITH vtab AS (INSERT INTO parent VALUES (6) RETURNING p_id)
 SELECT * FROM vtab

